bindings/python/iota_sdk/types/payload.py

```diff
@@ -3,50 +3,80 @@
 
 from __future__ import annotations
 from iota_sdk.types.common import HexStr
+from iota_sdk.types.output import Output
+from iota_sdk.types.input import UtxoInput
 from iota_sdk.types.signature import Ed25519Signature
-from dataclasses import dataclass
-from enum import Enum
+from iota_sdk.types.unlock import SignatureUnlock, ReferenceUnlock
+from dacite import from_dict
+from dataclasses import dataclass, field
+from enum import IntEnum
 from typing import Any, Optional, List
 
 
-class PayloadType(Enum):
+class PayloadType(IntEnum):
     TreasuryTransaction = 4
     TaggedData = 5
     Transaction = 6
     Milestone = 7
 
 
-class Payload():
-    def __init__(self, type, milestone: Optional[Any] = None, tagged_data=None,
-                 transaction=None, treasury_transaction: Optional[Any] = None):
-        """Initialize a payload
-        """
-        self.type = type
-        self.milestone = milestone
-        self.tagged_data = tagged_data
-        self.transaction = transaction
-        self.treasury_transaction = treasury_transaction
+@dataclass
+class TransactionEssence:
+    type: int
+
+
+@dataclass
+class RegularTransactionEssence(TransactionEssence):
+    networkId: str
+    inputsCommitment: HexStr
+    inputs: List[UtxoInput]
+    outputs: List[Output]
+    payload: Optional[TaggedDataPayload] = None
+    type: int = field(default=1, init=False)
 
     def as_dict(self):
         config = {k: v for k, v in self.__dict__.items() if v is not None}
 
-        if "milestone" in config:
-            del config["milestone"]
-        if "tagged_data" in config:
-            del config["tagged_data"]
-        if "transaction" in config:
-            del config["transaction"]
-        if "treasury_transaction" in config:
-            del config["treasury_transaction"]
+        if 'payload' in config:
+            config['payload'] = config['payload'].as_dict()
+
+        config['inputs'] = list(map(
+            lambda x: x.__dict__, config['inputs']))
 
-        config['type'] = config['type'].value
+        config['outputs'] = list(map(
+            lambda x: x.as_dict(), config['outputs']))
+
+        return config
+
+
+@dataclass
+class Payload():
+    """Initialize a Payload.
+    """
+    type: int
```

Comment:
So much better :)

Reply:
Yeah found the nice `field()` thing from dataclasses which makes this possible
Will update the remaining classes in another PR